components/modules/eromfs.c

```diff
@@ -0,0 +1,377 @@
+#include "module.h"
+#include "lauxlib.h"
+
+#include "esp_vfs.h"
+#include <errno.h>
+#include <dirent.h>
+
+#include <stdint.h>
+#include <string.h>
+
+/**
+ * The logical layout of the embedded volumes is
+ * [ volume record ]
+ * [ volume record ]
+ * ...
+ * [ index record in vol1 ]
+ * [ index record in vol1 ]
+ * ...
+ * [ file contents in vol1 ]
+ * [ file contents in vol1 ]
+ * ...
+ * [ index record in vol2 ]
+ * [ index record in vol2 ]
+ * ...
+ * [ file contents in vol2 ]
+ * [ file contents in vol2 ]
+ *
+ * Both the volume records and index records are variable length so as to not
+ * waste space in their name fields. Finding the start of the index records
+ * for a volume is by reading the offs(et) field in the volume record and
+ * jumping that many bytes forward from the start of the eromfs.bin data.
+ * Similarly, finding the file contents is by reading the index record's
+ * offs(et) field and basing that off the start of the volume index.
+ * Naturally, the start of the volume index is the same as the end of the
+ * volume header, and the start of the file contents is the same as the
+ * end of the volume index, and either of those can be worked out by
+ * reading the offs(et) field in the first record.
+ */
+
+#pragma pack(push, 1)
+typedef struct {
+  uint8_t  rec_len;
```

Comment:
If you moved the rec_len to be just before the name here and below, then you wouldn't need to do that pragma pack, and the unit16/unit32 would be aligned.

Reply:
Yes, but then I'd still have filler at the end, and I didn't want to waste space needlessly. If you have a volume with a large pack of small icon files the padding adds up.